src/components/ui/MonacoEditor.svelte

```diff
@@ -78,54 +81,84 @@
       ...(value !== null && { value }),
     };
     monaco = await import('monaco-editor');
-    monaco.languages.typescript.typescriptDefaults.setWorkerOptions({
-      customWorkerPath: '/customTS.worker.js',
-    });
     editor = monaco.editor.create(div, options, override);
 
-    editor.onDidChangeModelContent((e: Editor.IModelContentChangedEvent) => {
-      const newValue = editor?.getModel()?.getValue();
-      if (newValue !== undefined) {
-        dispatch('didChangeModelContent', { e, value: newValue });
-      }
-    });
-
-    // So.. there is no way to check when the model is initialized apparently!
-    // https://github.com/microsoft/monaco-editor/issues/115
-
-    // If we accidentally call the `getTypeScriptWorker()` function to early, it throws.
-    // Just use retry with exponential back-off to get it!
-    promiseRetry(
-      async () => {
-        let tsWorker: TypeScriptWorker | null = null;
-
-        // Errors in this block indicate failure to find a loaded worker
-        // so transform to the specific error type we care about.
-        try {
-          const getWorker: ((...uris: Uri[]) => Promise<TypeScriptWorker>) | undefined =
-            await monaco?.languages.typescript.getTypeScriptWorker();
-          if (getWorker) {
-            tsWorker = await getWorker();
-          }
-        } catch (e) {
-          throw new ShouldRetryError();
+    if (language && language === 'typescript') {
+      monaco.languages.typescript.typescriptDefaults.setWorkerOptions({
+        customWorkerPath: '/customTS.worker.js',
+      });
+
+      codeActionProvider = monaco.languages.registerCodeActionProvider(language, {
+        provideCodeActions: (model, range, context, token) => {
+          // create quick-action to convert unbalanced time to balanced time.
+          const codeActions = context.markers
+            .filter(error => error.code === String(CustomErrorCodes.Type.UNBALANCED))
+            .map(unbalancedTime => {
+              const match = unbalancedTime.message.match(/Suggestion:\s*(.*)/);
```

Comment:
I'm wondering if it makes sense to put this code action provider in the MonacoEditor component? Do we need this for all editors? For example does it make sense to check for unbalanced times in the scheduling editor?

Reply:
Good point, I made things more modular. You can now specify a `provideCodeActions` to a **MonacoEditor** if you want to use the quick action feature.

Reply:
Does the lint error still occur if you add an underscore to the parameter name?